src/pykeen/nn/modules.py

```diff
@@ -2549,17 +2550,77 @@ def forward(
+    def __init__(
+        self,
+        clamp_score: Clamp | float | None = None,
+        base: HintOrType[
+            Interaction[HeadRepresentation, RelationRepresentation, TailRepresentation]
+        ] = DistMultInteraction,
+        base_kwargs: OptionalKwargs = None,
+    ):
+        """
+        Initialize the interaction module.
+
+        :param clamp_score:
+            whether to clamp scores into a fixed interval
+        :param base:
+            the base interaction. Defaults to :class:`~pykeen.nn.modules.DistMultInteraction`.
+        :param base_kwargs:
+            keyword-based parameters used to instantiate the base interaction
+        """
+        super().__init__()
+        if isinstance(clamp_score, float):
+            clamp_score = (-clamp_score, clamp_score)
+        self.clamp_score = clamp_score
+        self.base = interaction_resolver.make(base, base_kwargs)
+
+    # docstr-coverage: inherited
+    def forward(self, h: HeadRepresentation, r: RelationRepresentation, t: TailRepresentation) -> FloatTensor:
+        scores = self.base(h, r, t)
+        if self.clamp_score is None:
+            return scores
+        low, high = self.clamp_score
+        return torch.clamp(scores, min=low, max=high)
+
+
 @parse_docdata
 class SimplEInteraction(
-    FunctionalInteraction[
-        tuple[FloatTensor, FloatTensor],
-        tuple[FloatTensor, FloatTensor],
-        tuple[FloatTensor, FloatTensor],
-    ],
+    Interaction[tuple[FloatTensor, FloatTensor], tuple[FloatTensor, FloatTensor], tuple[FloatTensor, FloatTensor]],
 ):
-    """A module wrapper for the SimplE interaction function.
+    r"""The SimplE interaction function.
+
+    SimplE can be regarded as extension of (a special case of) :class:`pykeen.nn.modules.CPInteraction`,
+    an early tensor factorization approach in which each entity
+    $e \in \mathcal{E}$ is represented by two vectors $\mathbf{e}_h, \mathbf{e}_t \in \mathbb{R}^d$ and each
+    relation by a single vector $\mathbf{r} \in \mathbb{R}^d$. Depending whether an entity participates in a
+    triple as the head or tail entity, either $\mathbf{e}_h$ or $\mathbf{e}_t$ is used. Both entity
+    representations are learned independently, i.e. observing a triple $(h,r,t)$, the method only updates
+    $\mathbf{h}_h$ and $\mathbf{t}_t$.
+    In contrast to :class:`~pykeen.nn.modules.CPInteraction`, SimplE introduces separate weights for each relation:
+    $\textbf{r}_{\rightarrow}$ and $\textbf{r}_{\leftarrow}$ for the inverse relation.
+    The interaction model is based on both:
 
-    .. seealso:: :func:`pykeen.nn.functional.simple_interaction`
+    .. math::
+
+        \frac{1}{2}\left(
+              \left\langle\textbf{h}_{h}, \textbf{r}_{\rightarrow}, \textbf{t}_{t}\right\rangle
+            + \left\langle\textbf{t}_{h}, \textbf{r}_{\leftarrow}, \textbf{h}_{t}\right\rangle
+        \right)
```

Comment:
It maybe makes sense to add some note there that this implementation is more general in that it allows chosing `base`s different to `DistMult`. Could also make sense to rename this class to something like `DirectionAverageInteraction` or similar, and create a specific sub-class / instantiation just for Simple (with a non-configurable default of `base=DistMult`). 🤔

Reply:
Yes I like the idea of making a generic base class and having simple be a special case

Reply:
done in https://github.com/pykeen/pykeen/pull/1470/commits/fe9ced2b7bdf1aa91b84f380b680164854606b18